### ISSUE TITLE
Reorganize model code into folders

### ISSUE THREAD

**Issue by jbischof** [keras-team/keras-nlp#365]
As we add more functionality per model, our `models` directory will be easier to read with more structure. Instead of long filenames in a flat structure like this:
```
models/
   __init__.py
   bert.py
   bert_tasks.py
   bert_vocabularies.json
   bert_checkpoints.json
```

We could have short filenames in a nested structure:
```
models/
   __init__.py
   bert/
      __init__.py
      model.py
      tasks.py
      vocabularies.json
      checkpoints.json
```

**Comment by mattdangerw** [comment]
I agree that having a nested directory structure is nice!

It seems like we could try to optimize our file structure around pieces that can be updated separately by different contributors. Preprocessing and graph code comes to mind here. Should we split files there?

Re-short names, I actually prefer the model name in the title, because it allows for better quick lookup of files in IDEs and filter searching my filename. Same reason we avoid `utils.py`. But that's a minor point.

Re-json, I think we are probably better off going with python files containing json like data structures. The main advantage to python is better integration with keras.io. Our documentation rendering is pretty heavily reliant on the ability to traverse python symbols. What is the advantage to json over an equivalently named and structured python file?

**Comment by jbischof** [comment]
Thanks for the feedback @mattdangerw. I think if we're unsure about json and short names I'd put this at lower priority. I'll keep the issue open for now so we can revisit.

**Comment by mattdangerw** [comment]
Short names is potato, potahto, I would say. Let's just choose one and roll with it.

Json vs python I think is one of those places where in Keras as a whole we would favor python, and it would integrate better with our tooling. I don't see the pro for having a json vs an equivalently structured standalone python file. That point might be worth quickly touching on in a meeting sometime.

**Comment by mattdangerw** [comment]
Could we consider something like...

```
__init__.py
bert_metadata.py.  # URLs to checkpoints and checkpoint metadata.
bert_models.py.  # Backbones.
bert_preprocessing.py. # Preprocessing layers.
bert_tasks.py  # Head wrappers.
```

I'm trying to basically split along lines of independent work here to the degree we can.
 * Allows graph code to come in separate from preprocessing code (this has already happened for us a few times).
 * Code for providing heads can come in later on top.
 * Checkpoints and vocabs can come in later on top (though it will inevitably hit all these files I think).

**Comment by jbischof** [comment]
This seems similar to the original proposal but with long names. 

* Won't checkpoints and vocabs go in `bert_metadata.py` as described in your comment?
* Where would bert-specific pipelines (a la #346) go?

**Comment by mattdangerw** [comment]
> This seems similar to the original proposal but with long names.

Yeah, sorry maybe should have done short names!

The main difference was to split the preprocessing and modeling into separate files, and merge the vocab and checkpoint metadata into a single file.

> Won't checkpoints and vocabs go in bert_metadata.py as described in your comment?

Yes, assuming we go with ID approach, I think splitting a vocab metadata file from a model metadata file is the wrong call. The vocab URL should live right with the model weights for `bert_uncased_en` say.

> Where would bert-specific pipelines

bert_tasks I guess? Haven't don't too much thinking there. The main updates to your proposal were to split model and preprocessing, and consolidate metadata per "checkpoint ID" in a single file.

**Comment by mattdangerw** [comment]
Synced offline, will go with

```
checkpoints.py
models.py
preprocessing.py
tasks.py
```

I'm going to toss up a PR just doing the file moving, because I think this will only get harder.
